### PR TITLE
Fix file system resize script hang

### PR DESCRIPTION
**Pull request by mglae**
## Bug
When resizing the storage file system during first boot the process stops while showing the "Checking layout..." spinner. After forcing a reboot LE starts with a 27MB storage partition.

The error can easily be reproduced by using the **run** option of a new created USB stick. I found it using LE 7.95.3 Generic.

## Error description
Parted was used in interactive mode with parameter **print fix** to move the backup GPT header to end of disk.
- There is no **fix** parameter in parted.
- Being interactive parted waits for user confirmation before fixing the GPT.
- More bad, parted may request more than one user interaction, i.e. when there is an old not matching GPT found at the end of the medium.
- There is no way to fix the GPT in parted batch mode.

## Fix
Create the new gptfdisk package to install sgdisk and use **sgdisk -e** to move the GPT. The solution is tested successfully with the LE8 branch.

### CODE REVIEW

**Comment by lrusak** [comment]
Thanks! Looks pretty good to me. I'll have to give it a test.

**Comment by stefansaraev** [comment]
:)

**Comment by stefansaraev** [comment]
however, I have a question. how does sgdisk work on mbr partitioned sd cards ?

File: packages/sysutils/gptfdisk/package.mk [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+PKG_DEPENDS_TARGET="toolchain"
+PKG_BUILD_DEPENDS_TARGET="toolchain"
+PKG_SECTION="system"
+PKG_SHORTDESC="GPT text-mode partitioning tools"
+PKG_LONGDESC="GPT text-mode partitioning tools"
+PKG_IS_ADDON="no"
+PKG_AUTORECONF="no"
+
+make_target() {
+  make sgdisk
```
Comment (vpeter4) [inline packages/sysutils/gptfdisk/package.mk]: Pass CC/CXX here and remove the patch file?

**Comment by chewitt** [comment]
https://bugs.launchpad.net/ubuntu/+source/parted/+bug/1490608 hints a way to fake tty input allowing `parted print fix` to work.

**Comment by mglae** [comment]
> however, I have a question. how does sgdisk work on mbr partitioned sd cards ?

Detects the situation and aborts. Example using a stick:
```
> sgdisk -e /dev/sdb

***************************************************************
Found invalid GPT and valid MBR; converting MBR to GPT format
in memory.
***************************************************************


Warning! Secondary partition table overlaps the last partition by
33 blocks!
You will need to delete this partition or resize it in another utility.
Non-GPT disk; not saving changes. Use -g to override.
```

**Comment by mglae** [comment]
> https://bugs.launchpad.net/ubuntu/+source/parted/+bug/1270203/comments/1 hints a way to fake tty input allowing parted print fix to work.

The problem is not the tty input but the not predictable behavior of `parted print` because of the target medium data structures.

**Comment by MilhouseVH** [comment]
> The problem is not the tty input but the not predictable behavior of parted print because of the target medium data structures.

Aren't these the target medium data structures that we have created?

**Comment by stefansaraev** [comment]
> Aren't these the target medium data structures that we have created?

not really. you only rewrite first xx megabytes of user's usb stick, what would parted do if it finds an old backup header at the end of the disk? does it print a warning, asks for input or rewrites it silently?

it's just an example that you have no control of what users have on their usb sticks and maybe have no idea how parted behaves.

File: packages/sysutils/gptfdisk/package.mk (hunk 1 above)
Comment (mglae) [inline packages/sysutils/gptfdisk/package.mk]: Yes, simplifies it.

**Comment by vpeter4** [comment]
What about just cleaning few MB at the end of partition before resizing?

**Comment by stefansaraev** [comment]
> What about just cleaning few MB at the end of partition before resizing?

if you want to properly support live mode on x86 instead of doing fragile workarounds, this PR is the way to go.

**Comment by mglae** [comment]
Sorry, still learning github - will be fixed soon.

**Comment by chewitt** [comment]
An approach proven in my old AppleTV installer script is to create a new partition table on-top of the old one. Then create two new partitions. As long as the first exactly matches the geometry of the original partition the existing filesystem (and contents) remains intact. Then create a second sized to `100%` to fill the disk, and format. The result gives correct backup tables at end of disk and intermediate locations. The only parted errors you see in this process are due to errors in the physical media and you can || abort/fail appropriately.

**Comment by escalade** [comment]
gptfdisk depends on popt.

**Comment by mglae** [comment]
> gptfdisk depends on popt.

Is added, thanks.

**Pull request merged**